### ISSUE TITLE
Bug: `MethodWhitelist` does not accept constructors

### ISSUE THREAD

**Issue by patrickkusebauch** [pepakriz/phpstan-exception-rules#140]
It is needed for PHP build-in DateTime/DateTimeImmutable/DateInterval classes, that throw exceptions for invalid inputs. Furthermore, this extension will mark them even if the input is hardcoded.

**Comment by pepakriz** [comment]
Yeah, this feature is missing because phpstan-dynamic-extensions didn't support constructors then. And I think the situation is still the same, isn't it?
